connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -545,6 +545,67 @@ else if (!configState.connectors().contains(connName))
     }
 
     @Override
+    public synchronized void restartConnector(final String connName, final Callback<Void> callback) {
+        addRequest(new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                if (!configState.connectors().contains(connName)) {
+                    callback.onCompletion(new NotFoundException("Unknown connector: " + connName), null);
```

Comment:
Makes sense. This brings up an interesting point. Since that call to `checkConfigSynced` compares the config offset with the offset from the current assignment, calling `pause()` in the other patch will cause the check to fail (since it writes a new record to the config topic). That means we either need to change that check to allow inequality or we need to allow pause to rebalance the group.